apps/skilavottord/ws/src/app/services/icelandicTransportAuthority.services.ts

```diff
@@ -70,16 +73,20 @@ export class IcelandicTransportAuthorityServices {
       restPassword,
     )
 
-    // Convert the query parameters to a query string
-    const queryString = Object.entries(queryParams)
-      .map(
-        ([key, value]) =>
-          `${encodeURIComponent(key)}=${encodeURIComponent(value)}`,
-      )
-      .join('&')
+    let fullUrl = restURL
+
+    if (queryParams) {
+      // Convert the query parameters to a query string
+      const queryString = Object.entries(queryParams)
+        .map(
+          ([key, value]) =>
+            `${encodeURIComponent(key)}=${encodeURIComponent(value)}`,
+        )
+        .join('&')
```

Comment:
Can't this be done by doing `const queryString = new URLSearchParams(queryParams)` and then `queryString.toString()`?

Reply:
Fixed